consensus/validator.go

```diff
@@ -26,80 +24,7 @@ func (consensus *Consensus) onAnnounce(msg *msg_pb.Message) {
 		return
 	}
 
-	// verify validity of block header object
-	// TODO: think about just sending the block hash instead of the header.
-	encodedHeader := recvMsg.Payload
-	header := new(block.Header)
-	if err := rlp.DecodeBytes(encodedHeader, header); err != nil {
-		consensus.getLogger().Warn().
-			Err(err).
-			Uint64("MsgBlockNum", recvMsg.BlockNum).
-			Msg("[OnAnnounce] Unparseable block header data")
-		return
-	}
-
-	if recvMsg.BlockNum < consensus.blockNum ||
```

Comment:
why is block num verification also discarded? Then what's left in the announce?

Reply:
In unpushed code, I just didn't push yet. Have some more on the way